apm-lambda-extension/Makefile

```diff
@@ -1,31 +1,103 @@
-build:
+SHELL = /bin/bash -eo pipefail
+
+AWS_FOLDER = .aws
+
 ifndef GOARCH
 	GOARCH=amd64
 endif
+
+# Transform GOARCH into the architecture of the extension layer
+ifeq ($(GOARCH),amd64)
+	ARCHITECTURE=x86_64
+else
+	ARCHITECTURE=arm64
+endif
+
+export AWS_FOLDER GOARCH ARCHITECTURE
+
+build:
 	GOOS=linux go build -o bin/extensions/apm-lambda-extension main.go
 	chmod +x bin/extensions/apm-lambda-extension
-build-and-publish:
-ifndef AWS_DEFAULT_REGION
-	$(error AWS_DEFAULT_REGION is undefined)
-endif
+build-and-publish: validate-layer-name validate-aws-default-region
 ifndef AWS_ACCESS_KEY_ID
 	$(error AWS_ACCESS_KEY_ID is undefined)
 endif
 ifndef AWS_SECRET_ACCESS_KEY
 	$(error AWS_SECRET_ACCESS_KEY is undefined)
-endif
-ifndef ELASTIC_LAYER_NAME
-	$(error ELASTIC_LAYER_NAME is undefined)
 endif
 	GOARCH=${GOARCH} make build
 	GOARCH=${GOARCH} make zip
-	aws lambda publish-layer-version --layer-name "${ELASTIC_LAYER_NAME}" --zip-file "fileb://./bin/extension.zip"
+	$(MAKE) publish
 zip:
 	cd bin && rm -f extension.zip || true && zip -r extension.zip extensions && cp extension.zip ${GOARCH}.zip
 test:
 	go test extension/*.go -v
 env:
-ifndef GOARCH
-	export GOARCH=amd64
-endif
 	env
+dist: validate-branch-name build test zip
+	@cp ./bin/$(GOARCH).zip bin/$(BRANCH_NAME)-linux-$(GOARCH).zip
+
+# List all the AWS regions
+get-all-aws-regions:
+	@aws \
+		ec2 \
+		describe-regions \
+		--region us-east-1 \
+		--output json \
+		--no-cli-pager \
+		| jq -r '.Regions[].RegionName' > .regions
+
+# Publish the given LAYER in all the AWS regions
+publish-in-all-aws-regions: validate-layer-name get-all-aws-regions
+	@mkdir -p $(AWS_FOLDER)
+	@while read AWS_DEFAULT_REGION; do \
+		echo "publish '$(ELASTIC_LAYER_NAME)' in $${AWS_DEFAULT_REGION}"; \
+		AWS_DEFAULT_REGION="$${AWS_DEFAULT_REGION}" ELASTIC_LAYER_NAME=$(ELASTIC_LAYER_NAME) $(MAKE) publish > $(AWS_FOLDER)/$${AWS_DEFAULT_REGION}; \
+	done <.regions
+	$(MAKE) create-arn-file
+
+# Publish the given LAYER in the given AWS region
+publish: validate-layer-name validate-aws-default-region
```

Comment:
This goal is uses in two different places:
- The traditional one when using `./elastic-lambda.js install`
- The one to rollout the ARN in all the AWS regions, see `publish-in-all-aws-regions`

I decided to refactor `build-and-publish` in favour of avoiding running `aws lambda` command in two different Make goals.